webnn/resources/utils.js

```diff
@@ -653,6 +654,22 @@ const buildGemm = (operationName, builder, resources) => {
   return namedOutputOperand;
 };
 
+const buildLayerNorm = (operationName, builder, resources) => {
+  // MLOperand layerNormalization(MLOperand input, optional MLLayerNormalizationOptions options = {});
+  const namedOutputOperand = {};
+  const inputOperand = createSingleInputOperand(builder, resources);
+  const layerNormOptions = {...resources.options};
+  if (layerNormOptions.scale) {
+    layerNormOptions.scale = createConstantOperand(builder, layerNormOptions.scale);
+  }
+  if (layerNormOptions.bias) {
+    layerNormOptions.bias = createConstantOperand(builder, layerNormOptions.bias);
+  }
```

Comment:
> There's a below error:  "NotSupportedError: DirectML: layerNormalization: The scale and bias must be both given or not given."

Ah, scale and bias are not independently optional, as you must provide *both* of them or *neither* of them to DML FL 4.0 (https://learn.microsoft.com/en-us/windows/win32/api/directml/ns-directml-dml_mean_variance_normalization_operator_desc "ScaleTensor - If DML_FEATURE_LEVEL is less than DML_FEATURE_LEVEL_5_2, then this tensor is required if BiasTensor is present."). This was relaxed in DML feature level 5.2+, but our target DML FL 4.0 still has this restriction.

🤔 So maybe the Chromium WebNN backend needs to create a constant tensor full of zeros under the hood if bias is not given. @huningxin, what do you think about that?

Reply:
> 🤔 So maybe the Chromium WebNN backend needs to create a constant tensor full of zeros under the hood if bias is not given. @huningxin, what do you think about that?

It sounds good to me. Thanks @fdwr !

WPT test cases should follow the spec that treat scale and bias are independently optional. We can file an issue against the current Chromium implementation for the NotSupportedError due to DML FL 4.0 restriction.

Reply:
Opened Chromium issue https://bugs.chromium.org/p/chromium/issues/detail?id=1513041, please take a look, thanks.